src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -619,7 +620,9 @@ public void Activate()
 
         public IPopupImpl CreatePopup() => Win32Platform.UseOverlayPopups ? null : new PopupImpl(this);
 
-        public void Dispose()
+        public void Dispose() => Dispose(isBeingDestroyed: false);
+
+        private void Dispose(bool isBeingDestroyed)
```

Comment:
This could be confused with the common `private void Dispose(bool isDisposing)` pattern. I'd suggest to rename this method. Also, `isBeingDestroyed` could be `isFromWmDestroy` for clarification.

Reply:
Agreed, that might lead to confusion. So you'd suggest renaming `Dispose(bool isBeingDestroyed)` entirely to something along the line of `FreeResources(bool isFromWmDestroy)`?